resources/api/openapi.yml

```diff
@@ -201,54 +233,46 @@ components:
         ends_at:
           $ref: '#/components/schemas/DateTime'
         location:
-          $ref: '#/components/schemas/Location'
+          $ref: '#/components/schemas/Reference'
         shift_type:
-          $ref: '#/components/schemas/ShiftType'
+          $ref: '#/components/schemas/Reference'
         created_at:
           $ref: '#/components/schemas/DateTimeOptional'
         updated_at:
           $ref: '#/components/schemas/DateTimeOptional'
-        entries:
+        angel_types:
           type: array
-          description: Can be empty (for example on Schedule import of unused room)
+          description: Needed angel types and user entries, can be empty (for example on Schedule import of unused room)
           items:
-            $ref: '#/components/schemas/ShiftEntry'
+            $ref: '#/components/schemas/NeededAngelType'
         url:
           type: string
           example: https://example.com/shifts/42
           description: Direct link to the shift
       required:
         - id
-        - title
+        - name
         - description
         - starts_at
         - ends_at
         - location
         - shift_type
         - created_at
         - updated_at
-        - entries
+        - angel_types
         - url
     ShiftEntry:
       type: object
       properties:
-        users:
-          type: array
-          items:
-            $ref: '#/components/schemas/User'
-        type:
-          $ref: '#/components/schemas/AngelType'
-        needs:
-          type: integer
-          description: >
-            Number of users needed for the shift of the given type.
-            Will be more than users count when not full, less when overbooked
-            or 0 when only additional users with given type have been added.
-          example: 3
+        user:
+          $ref: '#/components/schemas/Reference'
+        freeloaded:
+          type: boolean
+          example: false
+          description: True if the user did not arrive to the shift
```

Comment:
I would prefer a more precise description like "True if the user was marked as freeloaded for not arriving to the shift"?